apps/graphql/src/apps/itinerary/dataloaders/ItineraryValidity.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+
+import stringify from 'json-stable-stringify';
+import qs from 'querystring';
+import { OptimisticDataloader } from '@kiwicom/graphql-utils';
+
+import fetch from '../../../services/fetch/Fetch';
+import type {
+  ItineraryCheckParameters,
+  ItineraryValidity,
+  ValidityApiResponseType,
+  ItinerariesType,
+} from '../Itinerary';
+
+export const parseParameters = (input: ItineraryCheckParameters) => {
+  const passengersTotal =
+    (input.passengers.adults ?? 0) +
+    (input.passengers.children ?? 0) +
+    (input.passengers.infants ?? 0);
+
+  const params = {
+    booking_token: input.bookingToken,
+    bnum: input.bags,
+    pnum: passengersTotal,
+  };
+  return params;
+};
+
+const fetchItineraryValidity = async (
+  parameters: $ReadOnlyArray<ItineraryCheckParameters>,
+) => {
+  const results: $ReadOnlyArray<ValidityApiResponseType> = await Promise.all(
+    parameters.map(params => {
+      return fetch(
+        `/v2/booking/check_flights?${qs.stringify(parseParameters(params))}`,
+      );
+    }),
+  );
+  return results.map(res => {
+    return sanitizeValidity(res);
+  });
+};
+
+const sanitizeValidity = (
+  response: ValidityApiResponseType,
+): ItineraryValidity => {
+  return {
+    itineraryChecked: response.flights_checked,
+    itineraryValid: !response.flights_invalid,
+    itinerary: sanitizeItinerary(response),
+  };
+};
+
+const sanitizeItinerary = (
+  response: ValidityApiResponseType,
+): ItinerariesType => {
+  // @TODO - map remaining (null) data
+  return {
+    id: mapItineraryId(response.flights),
+    type: null,
+    bookingToken: response.booking_token,
+    startTime: null,
+    endTime: null,
+    origin: null,
+    destination: null,
+    sectors: null,
+    price: {
+      currency: 'EUR',
+      amount: response.total,
+    },
+  };
+};
+
+const mapItineraryId = flights => {
```

Comment:
Be more generic in naming because we must expect that TequilaAPI in future won't be only about flights.